core/src/com/unciv/logic/GameSaver.kt

```diff
@@ -66,4 +66,20 @@ class GameSaver {
     fun setGeneralSettings(gameSettings: GameSettings){
         getGeneralSettingsFile().writeString(json().toJson(gameSettings), false)
     }
+
+    fun autoSave(gameInfo: GameInfo, postRunnable: () -> Unit = {}) {
+        val gameInfoClone = gameInfo.clone()
+        kotlin.concurrent.thread {
+            // the save takes a long time (up to a second!) and we can do it while the player continues his game.
+            // On the other hand if we alter the game data while it's being serialized we could get a concurrent modification exception.
+            // So what we do is we clone all the game data and serialize the clone.
+            saveGame(gameInfoClone, "Autosave")
+
+            // do this on main thread
+            Gdx.app.postRunnable {
+                postRunnable()
```

Comment:
What is this for?

Reply:
This is useful if you want to update the ui *after* the saving is done. For example, the *next turn* button in `WorldScreen` is updated that way:
```
GameSaver().autoSave(gameInfo) {
  nextTurnButton.enable() // only enable the user to next turn once we've saved the current one
  updateNextTurnButton()
}
```